### PR TITLE
workflows: add more on-schedule workflows

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR adds 2 new workflows running on schedule:
- Action to label pull requests, only those adding new formula or removing an existing one for now
- Action to cancel previous workflow runs, so we don't have to do it manually and waste precious runners time

https://github.com/Homebrew/actions/pull/25
https://github.com/Homebrew/actions/pull/26

### CODE REVIEW

File: .github/workflows/autolabel.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Label
+        uses: Homebrew/action/label-pull-requests@master
+        with:
+          token: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+          def: |
+            [
+                {
+                    "label": "new formula",
```
Comment (SMillerDev) [inline .github/workflows/autolabel.yml:20]: Can we also do `marked for removal`

Comment (dawidd6) [reply .github/workflows/autolabel.yml:20]: Sure, added 👍

**Review by SMillerDev — DISMISSED** [review]

**Review by MikeMcQuaid — DISMISSED** [review]
Very nice 👏🏻

**Review by jonchang — APPROVED** [review]
Let's try it out!

File: .github/workflows/autocancel.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # Once every 15 minutes
+    - cron: '*/15 * * * *'
+
+jobs:
+  autocancel:
+    if: github.repository == 'Homebrew/homebrew-core'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel
+        uses: Homebrew/action/cancel-previous-runs@master
```
Comment (jonchang) [inline .github/workflows/autocancel.yml:14]:
```suggestion
        uses: Homebrew/actions/cancel-previous-runs@master
```

File: .github/workflows/autolabel.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    # Once every 30 minutes
+    - cron: '*/30 * * * *'
+
+jobs:
+  autolabel:
+    if: github.repository == 'Homebrew/homebrew-core'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Label
+        uses: Homebrew/action/label-pull-requests@master
```
Comment (jonchang) [inline .github/workflows/autolabel.yml:14]:
```suggestion
        uses: Homebrew/actions/label-pull-requests@master
```

**Review by jonchang — CHANGES_REQUESTED** [review]

**Pull request merged**